### DISCUSSION TITLE
sysrepo plugin for strongswan

### DISCUSSION THREAD

**Discussion by Ali-aqrabawi** [strongswan/strongswan#2054] [Ideas]
it's nice to have sysrepo plugin for strongswan, i'm creating one, please let me know if it can be part of strongswan repo, or shall i create it in my own repo.

**Comment by strongX509** [comment]
Have you already defined an IPsec/IKE YANG model that can be applied to strongSwan?

**Reply by Ali-aqrabawi** [reply]
There is no yang model for ipsec-ike. There is an experimental ietf one. I have created one for strongswan, and I'm working on the sysrepo plugin. Im just wondering if it can be part of strongswan project.

**Reply by tobiasbrunner** [reply]
There is standardized model (RFC 9061), why not use that? Writing a custom model makes no sense to me.

**Reply by tobiasbrunner** [reply]
Also, it might be an option to write a proxy/wrapper that translates between [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) and the YANG model.

**Reply by Ali-aqrabawi** [reply]
custom model will provide full support of `strongswan`, for instance, RFC 9061 does not provide ability to create `pools{}`. another option is to create an extension model for `stronswan`  out of `RFC 9061`, we can decide on this later, but my question if this `sysrepo` plugin can be part of strongswan's plugins? or shall i maintain it separately ?

**Reply by tobiasbrunner** [reply]
Why full support? In what scenarios/use cases is the YANG model even used in practice? And why a plugin? Seems that translating between YANG model and vici (e.g. in Python) would be simpler and could avoid redundancies.

> but my question if this `sysrepo` plugin can be part of strongswan's plugins?

Impossible to answer at this point. No idea what you are planning exactly, what the result would look like, or what its quality would be etc.

> or shall i maintain it separately ?

Maybe. But that's how the workflow would be anyway, develop something in a fork then maybe file a pull request.
